README.md

```diff
@@ -61,6 +59,190 @@ yarn start
 
 Finally, we can navigate a browser window to `localhost:8080` to view generated data.
 
+### Running with Docker
+
+You can build and run sourcecred in a container to avoid installing dependencies on your host. First, build the container:
+
+```bash
+$ docker build -t sourcecred/sourcecred .
+```
+
+If you want to build and customize the `SOURCECRED_DIRECTORY`, you can set that as a `--build-arg`:
+
+```bash
+$ docker build --build-arg SOURCECRED_DEFAULT_DIRECTORY=/tmp/data -t sourcecred/sourcecred .
+```
+
+Your options for running the container including the following commands. 
+Examples will be shown for each.
+
+ - **dev-preview**: offers a shortcut for loading sourcecred and then starting a dev server. This is likely the option you'll choose if you want to provide a respository or an organization and preview results a web interface.
+ - **dev-server**: exposes several webpack operations without the initial load. This takes no arguments.
+ - **build**: simply provides the build command to yarn, followed by any argumnents that you provide.
+ - **(anything else)**: will be passed on to sourcecred.js
+
+#### Development Preview 
+
+To run the development preview, you will still need to export a GitHub token, and then provide it to the container when you run it. 
+Notice that we are also binding port 8080 so we can view the web interface that will be opened up.  
+The only argument needed is a command to load the GitHub repository to generate the sourcecred for:
+
+```bash
+REPOSITORY=sfosc/sfosc
+$ SOURCECRED_GITHUB_TOKEN="xxxxxxxxxxxxxxxxx" \
+    docker run -d --name sourcecred --rm --env SOURCECRED_GITHUB_TOKEN -p 8080:8080 sourcecred/sourcecred dev-preview "${REPOSITORY}"
+```
+
+You can also specify an entire organization:
+
+```bash
+ORGANIZATION=@sfosc
+$ SOURCECRED_GITHUB_TOKEN="xxxxxxxxxxxxxxxxx" \
+    docker run -d --name sourcecred --rm --env SOURCECRED_GITHUB_TOKEN -p 8080:8080 sourcecred/sourcecred dev-preview "${ORGANIZATION}"
+```
+
+If you want to bind the data folder to the host, you can do that too. 
+In the example below, we have a folder "data" in the present working directory that we bind to "/data" in the container, the default `SOURCECRED_DIRECTORY`.
+
+```bash
+$ SOURCECRED_GITHUB_TOKEN="xxxxxxxxxxxxxxxxx" \
```

Comment:
The load command will terminate when done. So I would suggest not using `-d` but `-ti` instead.

Reply:
Also does not require the published port.

Reply:
No that's a typo that should be dev-preview too (not load). Sorry about that!

Reply:
Ah, I thought load + dev-server is a more natural example of when you'd use volumes, dev-preview being a bit more self-contained. But I'm not to picky about it either way. Planned behaviour would be more like dev-preview anyway.

Reply:
okay I just edited it to be (I think) in vision of what you want. If I did it wrong, please go ahead and fix it (because I might not be sure what you have in mind). I did my best!